### DISCUSSION TITLE
Go version too high

### DISCUSSION THREAD

**Discussion by mauza** [riverqueue/river#156] [General]
The company I work for is stuck for the foreseeable future on go 1.20. Is there a reason you are on go 1.21? I'm wondering if it is possible to support a lower go version.

**Comment by bgentry** [comment]
One specific reason is that we decided to settle on slog for logging. That got released in Go 1.21. It *might* be possible to remove this requirement, though I don’t think it will be a short term focus for us.  We also use some of the newer context APIs like cancel and timeout cause and will probably keep doing so.

We’re mostly working on rounding out the core feature set and improving reliability / fixing bugs at this early stage. Going forward I expect the project will always support at least the 2 most recent major versions, though we may support more than that if the tradeoff seems worthwhile.

Out of curiosity, what makes you stuck on the older version?

**Comment by brandur** [comment]
+1. Also worth remembering that Go 1.22 is slated for release next month. I don't think we'll need to jump to it immediately for anything, so that'll mean that River supports Go 1.21 and 1.22, which according to their versioning policy, will be all the officially supported versions of Go (two latest majors).

That said, I think we'd probably work towards a place where we're a little more generous with supported majors than the official language just because @mauza, I've been in a situation like you where it's unfortunately made difficult to upgrade for various organizational reasons.

**Comment by mauza** [comment]
Yeah, I'm pushing on the org to do the work so we can upgrade. In the mean time I forked and got it to work with 1.20. I appreciate the response. That sounds like an good path for the project.

**Comment by patryk4815** [comment]
In my case I can't use go1.21+, because Go 1.20 was last release for certain windows/macos version

**Reply by brandur** [reply]
Out of curiosity, which Windows/macOS version?

**Reply by patryk4815** [reply]
windows 2012 + 2008
